third_party/amd/lib/TritonAMDGPUTransforms/BlockPingpong.cpp

```diff
@@ -1126,6 +1125,49 @@ LogicalResult Pingponger::transformFP4(OpBuilder &builder, Location loc) {
   return success();
 }
 
+LogicalResult Pingponger::transformNS3(OpBuilder &builder, Location loc) {
```

Comment:
I see there's already a comment for documentation; but it would be great to capture what these transform functions do; e.g. what is the assumed state of the IR before tranformation, how/when is conditional barrier inserted, what the execution mode looks like before/after the transformation.